libraries/provider_environment.rb

```diff
@@ -41,12 +41,8 @@ def whyrun_supported?
       def object_template
         search_pattern = new_resource.search_pattern || "chef_environment:#{new_resource.environment}"
         if new_resource.limit_region && !new_resource.server_region
-          if node.key?('ec2')
-            server_region = node['ec2']['placement_availability_zone'].chop
-          else
-            # add more cloud providers
-            server_region = nil
-          end
+          server_region = nil
+          server_region node['ec2']['placement_availability_zone'].chop if node.key?('ec2')
```

Comment:
`new_resource.server_region` is ignored, its a pre existing bug in the condition and server_region is missing assignment operator. i will update the if block to -

```ruby
         search_pattern = new_resource.search_pattern || "chef_environment:#{new_resource.environment}"
-        if new_resource.limit_region && !new_resource.server_region
-          server_region = nil
-          server_region node['ec2']['placement_availability_zone'].chop if node.key?('ec2')
+        server_region = new_resource.server_region
+
+        if new_resource.limit_region && !server_region
+          server_region = node['ec2']['placement_availability_zone'].chop if node.key?('ec2')
         end
```